Lodgify.Payments.Stripe.Domain/Accounts/Account.cs

```diff
@@ -26,19 +29,53 @@ internal Account(Guid id) : base(id)
 
     public static Account Create(int userId, string email, string stripeAccountId, string controllerType, string losses, string fees, string requirementCollection, string dashboard, bool chargesEnabled, bool detailsSubmitted)
     {
-        return new Account()
+        var account = new Account(Guid.NewGuid())
         {
-            Id = Guid.NewGuid(),
             UserId = userId,
             Email = email,
             StripeAccountId = stripeAccountId,
             ControllerType = controllerType,
             Losses = losses,
             Fees = fees,
             RequirementCollection = requirementCollection,
-            Dashboard = dashboard,
-            ChargesEnabled = chargesEnabled,
-            DetailsSubmitted = detailsSubmitted
+            Dashboard = dashboard
         };
+
+        //verify this flag in case if stripe set chargesEnabled = true and detailsSubmitted = true direct after create account
+        if (chargesEnabled)
+        {
+            account.SetChargesEnabled(chargesEnabled, DateTime.UtcNow);
+        }
+
+        if (detailsSubmitted)
+        {
+            account.SetDetailsSubmitted(detailsSubmitted, DateTime.UtcNow);
+        }
+
+        return account;
+    }
+
+    public bool SetChargesEnabled(bool chargesEnabled, DateTime changeRequestedAt)
+    {
+        if ((!ChargesEnabledAt.HasValue || ChargesEnabledAt.Value < changeRequestedAt) && ChargesEnabled != chargesEnabled)
```

Comment:
As said in the previous comments I think `ChargesEnabledAt` (`ChargesEnabledSetAt`) should not be nullable
```suggestion
        if (ChargesEnabledAt.Value < changeRequestedAt && ChargesEnabled != chargesEnabled)
```